fuzzingandroid/FuzzerEngine/fuzzerengine/executor.py

```diff
@@ -277,21 +285,46 @@ def start_exec(self, log_proc, monkey_watcher, recent_path, recent_path_size):
     def bring_app_to_front(self,pkg_name):
 
         try:
-            cmd="adb -s " + vm.VM.ip+ ":"+vm.VM.adb_port +" shell dumpsys activity recents | grep realActivity="+ pkg_name + "  | cut -d'=' -f2 | xargs adb shell am start "
+            cmd="adb -s " + vm.VM.ip+ ":"+vm.VM.adb_port +" shell dumpsys activity recents | grep realActivity="+ pkg_name + "  | cut -d'=' -f2 | xargs adb -s " + vm.VM.ip+ ":"+vm.VM.adb_port + " shell am start "
             os.system(cmd + " > /dev/null 2>&1")
         except Exception, e:
             print "resumed app failed."
 
 
-    def init_app(self, app_name):
+    def init_app(self, app_name, login_script):
 
         print "launching app under test..."
-        os.system("adb shell monkey -p " + app_name + "  1") 
+        cmd="adb -s " + vm.VM.ip + ':' + vm.VM.adb_port + " shell monkey -p " + app_name + "  1"
+        print cmd
+        while True:
+            p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+            output = str(p.stdout.read()).strip()
+            if "No activities found to run" in output:
+                print "ERROR: launch failed! Try Again!"
+                time.sleep(5)
+            elif "Events injected" in output:
+                print "SUCCESS: app is launched!"
+                break
+            else:
+                print "New Message"
+                break
```

Comment:
Reconnect to ADB as long as connection closes due to "No activities found to run".
@Zhendong2050: Probably a bug in TM?